src/bin/tikv-server.rs

```diff
@@ -332,12 +333,25 @@ fn run_raft_server(listener: TcpListener, matches: &Matches, config: &toml::Valu
     let pd_client = Arc::new(new_rpc_client(&pd_addr, cluster_id).unwrap());
     let resolver = PdStoreAddrResolver::new(pd_client.clone()).unwrap();
 
-    let (store, raft_router) = build_raftkv(matches,
-                                            config,
-                                            ch,
-                                            cluster_id,
-                                            format!("{}", listener.local_addr().unwrap()),
-                                            pd_client);
+    let (store, raft_router, node_id) = build_raftkv(matches,
+                                                     config,
+                                                     ch,
+                                                     cluster_id,
+                                                     format!("{}", listener.local_addr().unwrap()),
+                                                     pd_client);
+
+    // append node id to metric prefix.
+    let mut prefix = get_string_value("metric-prefix",
+                                      "metric.prefix",
+                                      matches,
+                                      config,
+                                      Some("tikv".to_owned()),
+                                      |v| v.as_str().map(|s| s.to_owned()));
+    prefix.push_str(&format!(".{}", node_id));
+    if let Some(mut prefix_with_id) = config.lookup_mut("metric.prefix") {
+        *prefix_with_id = toml::Value::String(prefix);
+    }
+    initial_metric(matches, config);
```

Comment:
why not re-define the function like below:
```
initial_metric(matches, config, Option<Fn(prefix:&str)->String>

or 
initial_metric(matches, config, postfix: String>
```

so that you don't need find prefix first and then change the inner value.